daml-script/dump/src/main/scala/com/daml/script/dump/TreeUtils.scala

```diff
@@ -15,6 +17,21 @@ import scalaz.syntax.foldable._
 object TreeUtils {
   final case class Selector(i: Int)
 
+  /** Sort the active contract set topologically,
+    *  such that a contract at a given position in the list
+    *  is only referenced by other contracts later in the list.
+    */
+  def topoSortAcs(acs: Map[String, CreatedEvent]): List[CreatedEvent] = {
+    val graph: Graphs.Graph[String] = acs.mapValues(createdReferencedCids)
+    Graphs
+      .topoSort(graph)
+      .left
+      .map(msg => new RuntimeException(s"Encountered cyclic contract dependencies: $msg"))
+      .toTry
```

Comment:
this seems a bit odd, you first handle the `Left` case by throwing to an exception but then you convert it to a `Try` discarding the information that you already handled the `Left` and call a partial `Left`. Maybe a a pattern match or a fold would be more appropriate?

Reply:
Not quite, the `.left.map` converts the left type to an exception type and `toTry.get` then throws that in the left case. However, I agree it's too much acrobatics. I've changed it to a simple match.